ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/ledger_pruning_service.proto

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.ledger.api.v1.admin;
+
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "LedgerPruningServiceOuterClass";
+option java_package = "com.digitalasset.ledger.api.v1.admin";
+option csharp_namespace = "Com.DigitalAsset.Ledger.Api.V1.Admin";
+
+// Status: experimental interface, will change before it is deemed production
+// ready
+
+// Prunes/truncates the "oldest" ledger transactions from the ledger (specifically the Ledger Api Server) to
+// 1. keep the "inactive" portion of the ledger to a manageable size (i.e. removing the portion of the ledger that does
+//    not affect the set of allowed, future commands.
+// 2. remove inactive state to honor right-to-forget guarantees.
+service LedgerPruningService {
```

Comment:
Would it make sense to have a `pruneByOffset` and a `pruneByTime` method rather than using and relying on unions to discriminate?

Reply:
Sure, seems reasonable - will change.